core/src/io/anuke/mindustry/world/blocks/power/PowerGraph.java

```diff
@@ -205,8 +208,17 @@ public void update(){
                 powerProduced -= chargeBatteries(powerProduced - powerNeeded);
             }
         }
-
-        distributePower(powerNeeded, powerProduced);
+        if (Mathf.isZero(powerNeeded))
+        {
+        	return 0.0f;
```

Comment:
Why is this if-statement necessary? If no power is needed, the power balance should still be updated regardless.